clcache.py

```diff
@@ -74,10 +79,15 @@ def filesBeneath(path):
 
 
 def childDirectories(path, absolute=True):
-    for entry in os.listdir(path):
-        absPath = os.path.join(path, entry)
-        if os.path.isdir(absPath):
-            yield absPath if absolute else entry
+    for entry in LIST(path):
+        if type(entry) == str:
```

Comment:
Even if it turns out that we still need to support `os.listdir` for some reason (see my question above), then it's wasteful to check the type of every entry. We could just do the test once outside the loop via e.g. `if LIST == os.listdir`.